servers/rendering/shader_language.h

```diff
@@ -919,11 +919,14 @@ class ShaderLanguage {
 	bool check_warnings = false;
 	uint32_t warning_flags;
 
-	void _add_line_warning(ShaderWarning::Code p_code, const StringName &p_subject = "") {
-		warnings.push_back(ShaderWarning(p_code, tk_line, p_subject));
+	void _add_line_warning(ShaderWarning::Code p_code, const StringName &p_subject = "", const Variant &p_extra_arg1 = Variant(), const Variant &p_extra_arg2 = Variant()) {
+		warnings.push_back(ShaderWarning(p_code, tk_line, p_subject, p_extra_arg1, p_extra_arg2));
 	}
-	void _add_warning(ShaderWarning::Code p_code, int p_line, const StringName &p_subject = "") {
-		warnings.push_back(ShaderWarning(p_code, p_line, p_subject));
+	void _add_global_warning(ShaderWarning::Code p_code, const StringName &p_subject = "", const Variant &p_extra_arg1 = Variant(), const Variant &p_extra_arg2 = Variant()) {
+		warnings.push_back(ShaderWarning(p_code, -1, p_subject, p_extra_arg1, p_extra_arg2));
+	}
+	void _add_warning(ShaderWarning::Code p_code, int p_line, const StringName &p_subject = "", const Variant &p_extra_arg1 = Variant(), const Variant &p_extra_arg2 = Variant()) {
+		warnings.push_back(ShaderWarning(p_code, p_line, p_subject, p_extra_arg1, p_extra_arg2));
```

Comment:
I meant that these could also get a `p_extra_args` array now instead of limiting the amount.